src/System.Linq.Expressions/tests/ReadOnlyCollectionBuilder/ReadOnlyCollectionBuilder.Tests.IndexOf.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Tests;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests
+{
+    /// <summary>
+    /// Contains tests that ensure the correctness of the List class.
+    /// </summary>
+    public abstract partial class ReadOnlyCollectionBuilder_Generic_Tests<T> : IList_Generic_Tests<T>
+    {
+        #region Helpers
+
+        public delegate int IndexOfDelegate(ReadOnlyCollectionBuilder<T> list, T value);
+        public enum IndexOfMethod
+        {
+            IndexOf_T,
+        };
+
+        private IndexOfDelegate IndexOfDelegateFromType(IndexOfMethod methodType)
+        {
+            switch (methodType)
+            {
+                case (IndexOfMethod.IndexOf_T):
+                    return ((ReadOnlyCollectionBuilder<T> list, T value) => { return list.IndexOf(value); });
+                default:
+                    throw new Exception("Invalid IndexOfMethod");
+            }
+        }
+
+        /// <summary>
+        /// MemberData for a Theory to test the IndexOf methods for List. To avoid high code reuse of tests for the 6 IndexOf
+        /// methods in List, delegates are used to cover the basic behavioral cases shared by all IndexOf methods. A bool
+        /// is used to specify the ordering (front-to-back or back-to-front (e.g. LastIndexOf)) that the IndexOf method
+        /// searches in.
+        /// </summary>
+        public static IEnumerable<object[]> IndexOfTestData()
+        {
+            foreach (object[] sizes in ValidCollectionSizes())
+            {
+                int count = (int)sizes[0];
```

Comment:
nit: A slightly cleaner way to do it (what I usually do) is to add `System.Linq` and do

```cs
IEnumerable<int> counts = ValidCollectionSizes().Select(array => array[0]).Cast<int>();

foreach (int count in counts) { ... }
```